AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Utils/RESTOperationRequestUtils.swift

```diff
@@ -29,10 +29,11 @@ final class RESTOperationRequestUtils {
             components.path.append(path)
         }
 
-        if let queryParameters = queryParameters {
+        if let queryParameters = queryParameters?.sorted(by: { $0.key < $1.key }) {
```

Comment:
Reordering the query parameters is not necessarily an idempotent operation. Depending on the service implementation, `http://www.foo.com?foo[]=a&foo[]=b` may result in a different interpretation than `http://www.foo.com?foo[]=b&foo[]=a`. What is the reason for reordering if the intent is to simply encode the keys & values?

Reply:
The only reason for reordering is for making testing reliable as dictionaries don't guarantee any order on keys.
I also considered changing the method signature to accept [`KeyValuePairs`](https://developer.apple.com/documentation/swift/keyvaluepairs) instead of a `Dictionary` but it would have generated more changes

Reply:
As discussed, remove the reordering and change your test assertion to compare one parameter key/value pair at a time